### PR TITLE
Workaround for bug when AWS mistakenly set Content-Type to XML

### LINKED ISSUE #33964: Wrong s3 object content type for large files
Connected to ClickHouse server version 21.13.1 revision 54455.

```
INSERT INTO FUNCTION s3('https://s3.us-east-1.amazonaws.com/dzhuravlev/small_file.xml.gz', 'XML', 
'number UInt64', 'gz') SELECT * FROM numbers(10);

INSERT INTO FUNCTION s3('https://s3.us-east-1.amazonaws.com/dzhuravlev/large_file.xml.gz','XML', 
'number UInt64','gz') SELECT * from numbers(1e8);

aws s3api head-object --bucket  dzhuravlev  --key small_file.xml.gz
{
    "AcceptRanges": "bytes",
    "LastModified": "Mon, 24 Jan 2022 21:28:50 GMT",
    "ContentLength": 262,
    "ETag": "\"667796620f85cd6bad6957ea01d29548\"",
    "ContentType": "binary/octet-stream",
    "Metadata": {}
}

aws s3api head-object --bucket  dzhuravlev --key large_file.xml.gz
{
    "AcceptRanges": "bytes",
    "LastModified": "Mon, 24 Jan 2022 21:29:08 GMT",
    "ContentLength": 261909920,
    "ETag": "\"829dbc9697bcffd1528e8e6f55c564f4-8\"",
    "ContentType": "application/xml",                      ---<<<<<<<<<<<-- expected "ContentType": "binary/octet-stream",
    "Metadata": {}
}
```

So it seems that if the object size is bigger than 32MB then Clickhouse sets the wrong ContentType `application/xml` instead of `binary/octet-stream`.

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
When large files were written with `s3` table function or table engine, the content type on the files was mistakenly set to `application/xml` due to a bug in the AWS SDK. This closes #33964.

### CODE REVIEW

**Review by kitaisreal — APPROVED** [review]

**Comment by kitaisreal** [comment]
Need to fix flaky test 02207_s3_content_type.

**Review by alesapin — APPROVED** [review]

**Comment by alesapin** [comment]
Waiting only for flaky check.

**Comment by alesapin** [comment]
Finally!
```
02207_s3_content_type | OK | 1.85
-- | -- | --
```

File: tests/queries/0_stateless/02207_s3_content_type.sh [hunk 1]
```diff
@@ -10,15 +10,4 @@ $CLICKHOUSE_CLIENT --query "
 INSERT INTO TABLE FUNCTION s3('http://localhost:11111/test/content-type.csv.gz', 'test', 'testtest', 'CSV', 'number UInt64') SELECT number FROM numbers(1000000) SETTINGS s3_max_single_part_upload_size = 10000, s3_truncate_on_insert = 1;
 "
 
-# strace -e trace=network,write -s1000 -f ~/work/ClickHouse/mc cat clickminio/test/content-type.csv.gz >/dev/null
```
Comment (alexey-milovidov) [inline tests/queries/0_stateless/02207_s3_content_type.sh]: What was wrong with this command, just curious?

**Pull request merged**